lib/fetch/index.js

```diff
@@ -1344,7 +1344,7 @@ async function httpNetworkOrCacheFetch (
   //    user agents should append `User-Agent`/default `User-Agent` value to
   //    httpRequest’s header list.
   if (!httpRequest.headersList.contains('user-agent')) {
-    httpRequest.headersList.append('user-agent', 'undici')
+    httpRequest.headersList.append('user-agent', 'node')
```

Comment:
Is there any way we can see whether we are node internal or not? I'm a little off about doing this for the npm package.

Reply:
I agree. And I couldn't think of anything. Maybe someone else has an idea. Is there potentially some kinda `process.env` value?

Reply:
Maybe something like if __filename ends with index-fetch.js?

Reply:
How would we test that? Something in the Node.js side?

Reply:
And is `__filename` guaranteed to always be assigned? Regardless if the user's code is ESM or CJS for example?

Reply:
undici-fetch.js is the bundled build of undici that node core uses. I believe there are tests that use it so you'd test the fetch exported from there? I'm not home atm so I can't do any tests myself

Reply:
Looks like we were already building `undici-fetch.js` pre test, but it wasn't obvious another test was using it so I created a new test file.